### ISSUE TITLE
DiaNN (Ubuntu 22.04 using Wine 9.0) has err:sync:RtlpWaitForCriticalSection error

### ISSUE THREAD

**Issue by genesandbones** [vdemichev/DiaNN#920]
Hello,
I have searched around in the issues and I saw similar but not exactly the same issues and troubleshooting off of similar issues didn't seem to help with this issue.

I have DiaNN running via wine in ubuntu 22.04 and installed the correct thermofisher file.

From what I can tell, DiaNN does run but it seems to struggle with the loading of the file part and prints out the following error in the terminal:
`0510:err:sync:RtlpWaitForCriticalSection section 00006FFFFFCBE4A0 "dlls/ntdll/loader.c: loader_section" wait timed out in thread 0510, blocked by 0500, retrying (60 sec)`. And this will simply keep printing every minute unless I kill the processes.

In the DiaNN GUI, it is simply sitting at the loading part for loading in the file.

I have tried uninstalling and reinstalling various packages but have not figured out what is causing this threading issue. I'm running DiaNN as default as I'm just testing out its implementation on our computer with one .raw file.

**Comment by genesandbones** [comment]
After some further testing (and even trying v1.8 out) it seems to be related to having multiple input files.
I can run one sample no problem but if I add more than one then it becomes a problem.

Any thoughts/suggestions on this would be extremely helpful!
![image](https://github.com/vdemichev/DiaNN/assets/38194037/41902815-b10b-4bb6-b2b1-92375f5c4301)

Thanks!

**Comment by vdemichev** [comment]
Not within my expertise, but I would guess it fails to load the DLL. Is the Wine version 6.8 or later?
A workaround is to convert Thermo files to .dia (on Windows, with DIA-NN) or mzML (with whatever tool supports than on Linux).

**Comment by genesandbones** [comment]
Yeah, wine is 9.0
The workaround doesn't work either (converting to .dia). I can convert to .dia on a single file, but anytime I have more than one file in the list, it causes the thread error.

**Comment by nttg8100** [comment]
I set up the DIANN 1.8.1 using wine with MS Thermo Reader to simply run as command line and get similar issue. I solved this using below set up to build a docker image.
What I used is:
+ wine stable-9.0 based image 
+ install xvfb
+ wine32 to install diann 1.8.1
+ wine64 to install MS FileReader
+ commit this image to new docker image
+ run using wine inside a xvfb virtual screen

**Comment by KosSashaHelloWorld** [comment]
Have the same problem when trying launch RfaD (SkyrimSE mod) Wine 9.0. Clear Win10 prefix with only wine_mono_10 installed (and RfaD dists ofc):

> err:sync:RtlpWaitForCriticalSection section 00006FFFFFCBD4A0 "dlls/ntdll/loader.c: loader_section" wait timed out in thread 062c, blocked by 05c4, retrying (60 sec)

A lot of times (adds every min)

**Comment by vdemichev** [comment]
With DIA-NN 2.1.0 can access Thermo .raw natively on Linux, the Dockerfile is included in the distribution.

**Comment by KosSashaHelloWorld** [comment]
> With DIA-NN 2.1.0 can access Thermo .raw natively on Linux, the Dockerfile is included in the distribution.

May you please explain a bit more? I thought the main problem is thread (05c4) makes "critical section" busy and nobody (062c and others) can access it. I wish anybody may debug this and know why\what 05c4 waiting for, making 062c busy

P.S. help me (pls)

**Comment by KosSashaHelloWorld** [comment]
P.P.S. I mean is it possible to move the game to docker file (already compiled and patched ((mods)) ) with all the dlls and services needed?

> I guess not...

**Comment by KosSashaHelloWorld** [comment]
Btw thank you, Vadim. @vdemichev 

Best,
Alex.

**Comment by vdemichev** [comment]
No, unfortunately not possible to run Windows code under Docker on Linux.
